src/main/java/seedu/taassist/model/TaAssist.java

```diff
@@ -175,6 +181,51 @@ public void removeSession(ModuleClass moduleClass, Session session) {
         setModuleClass(moduleClass, moduleClass.removeSession(session));
     }
 
+    /**
+     * Exports the {@code moduleClass} from this {@code TaAssist} as a CSV file.
+     * {@code moduleClass} must exist in TA-Assist.
+     * @return The CSV file object created.
+     */
+    public List<List<String>> exportModuleClassToStringList(ModuleClass moduleClass) {
+        requireNonNull(moduleClass);
+
+        IsPartOfClassPredicate predicate = new IsPartOfClassPredicate(moduleClass);
+        List<Student> students = getStudentList().stream().filter(predicate).collect(Collectors.toList());
+
+        List<List<String>> fileData = new ArrayList<>();
+
+        // Header row
+        List<String> headerRow = new ArrayList<>();
+        headerRow.add(CSV_NAME_COLUMN_HEADER);
+        // Column header for each session
+        List<Session> sessions = moduleClass.getSessions();
+        for (Session s : sessions) {
+            headerRow.add(s.getSessionName());
+        }
+        fileData.add(headerRow);
+
+        // Row for each student
+        for (Student student : students) {
+            List<String> row = new ArrayList<>();
+            row.add(student.getName().toString());
+            Optional<StudentModuleData> moduleDataOptional = student.findStudentModuleData(moduleClass);
+            if (!moduleDataOptional.isPresent()) {
+                continue;
+            }
+            StudentModuleData moduleData = moduleDataOptional.get();
+            for (Session s : sessions) {
+                Optional<SessionData> sessionDataOptional = moduleData.findSessionData(s);
```

Comment:
Perhaps can consider rewriting the following block of code as such.
```java
row.add(sessionDataOptional
    .map(SessionData::getGrade)
    .map(String::valueOf)
    .orElseGet(CSV_EMPTY_GRADE));
```

Reply:
Note: This comment is not compiled, as such it might be wrong, make sure to manually verify this.

Reply:
Done in next commit.